spec.md

```diff
@@ -1657,40 +1629,35 @@ If a `CreateSnapshot` operation times out before the snapshot is cut, leaving th
-Uploading may be a long process that could take hours.
-If a `freeze` operation was done on the application before taking the snapshot, it could be a long time before the application can be running again if we wait until the upload is complete to `thaw` the application.
+Some cloud providers will process the snapshot after the snapshot is cut, i.e., uploading the snapshot to a location in the cloud (i.e., an object store) after the snapshot is cut.
+A process such as uploading may be a long process that could take hours.
+If a `freeze` operation was done on the application before taking the snapshot, it could be a long time before the application can be running again if we wait until the process is complete to `thaw` the application.
 The purpose of `freeze` is to ensure the application data is in consistent state.
 When `freeze` is performed, the container is paused and the application is also paused.
 When `thaw` is performed, the container and the application start running again.
-During the snapshot uploading phase, since the snapshot is already cut, a `thaw` operation can be performed so application can start running without waiting for the upload to complete.
-The status of the snapshot will become `READY` after the upload is complete.
+During the snapshot processing phase, since the snapshot is already cut, a `thaw` operation can be performed so application can start running without waiting for the process to complete.
+The `is_ready_to_use` parameter of the snapshot will become `true` after the process is complete.
 
-For cloud providers and storage systems that don't have the uploading process, the status should be `READY` after the snapshot is cut.
-`thaw` can be done when the status is `READY` in this case.
+For cloud providers and storage systems that don't have the process, the `is_ready_to_use` parameter should be `true` after the snapshot is cut.
+`thaw` can be done when the `is_ready_to_use` parameter is `true` in this case.
 
-A `CREATING` status is not included here because CreateSnapshot is synchronous and will block until the snapshot is cut.
+If the cloud provider or storage system needs to process the snapshot after the snapshot is cut, the `is_ready_to_use` parameter returned by CreateSnapshot SHALL be `false`.
+CO MAY continue to call CreateSnapshot while waiting for the process to complete until `is_ready_to_use` becomes `true`.
+Note that CreateSnapshot no longer blocks after the snapshot is cut.
 
-`ERROR` is a terminal snapshot status.
-A CO SHOULD explicitly delete snapshots in this status.
+A gRPC error code SHALL be returned if an error occurs during any stage of the snapshotting process.
+A CO SHOULD explicitly delete snapshots when an error occurs.
 
-The SnapshotStatus parameter provides guidance to the CO on what action can be taken in the process of snapshotting.
+The `is_ready_to_use` parameter provides guidance to the CO on what action can be taken in the process of snapshotting.
```

Comment:
Move this before line 1647.
Change `on what action can be taken` -> `on when it can "thaw" the application`

Reply:
Done